index.mjs

```diff
@@ -0,0 +1,165 @@
+//. # Fluture Fastify
+//.
+//. Create [Fastify][] handlers using [Future][]s from [Fluture][].
+
+import Daggy from 'daggy';
+import {isFuture} from 'fluture';
+import curry from 'lodash.curry';
+
+//. ## Usage
+//.
+//. ```console
+//. npm install --save fluture fluture-fastify
+//. ```
+//.
+//. Allows using pure functions as [Fastify][] handlers.
+//. Expects actions to return [Future][]s and the reply
+//. to be decorated with a property `locals`
+//. that will be passed to every action as its second argument.
+//.
+//. ```javascript
+//. import {handler, Typed} from 'fluture-fastify';
+//. import {after} from 'fluture';
+//. import createApp from 'fastify';
+//.
+//. const app = createApp({logger: true});
+//.
+//. const Json = Typed ('application/json; charset=utf-8');
+//. const action = (req, {number}) => after (100, Json (200, number));
+//.
+//. app.decorateReply ('locals', {number: 42});
+//. app.get ('/number', handler ('getNumber', action));
+//.
+//. app.listen (3000, '0.0.0.0');
+//. ```
+//.
+//. ## API
+//.
+//. ### Responses
+//.
+//# Typed :: (Str, Num, Any) -> Response a
+//.
+//. A typed response requires status code, some content and the
+//. content type. The type defines how the value is serialized.
+//. In order to see all the possible combinations see [reply.send][] docs.
+//.
+//. #### Plain text
+//. ```javascript
+//. const Plain = Typed ('text/plain; charset=utf-8');
+//.
+//. Plain (200, 'Number 42');
+//. ```
+//. #### Json
+//.
+//. ```javascript
+//. const Json = Typed ('application/json; charset=utf-8');
+//.
+//. Json (200, {number: 42});
+//. ```
+//. #### Stream
+//.
+//. ```javascript
+//. const Stream = Typed ('application/octet-stream');
+//.
+//. Stream (200, createReadStream ('file', 'utf8'));
+//. ```
+//# Serialized :: ((b -> a), Str, Num, b) -> Response a
+//.
+//. A typed response with a custom serializer.
+//.
+//. ```javascript
+//. const Proto = Serialized (protoBuf.serialize, 'application/x-protobuf');
+//.
+//. Proto (200, new protoBuf ());
+//. ```
+//# Redirect :: (Num, Str) -> Response a
+//.
+//. A redirection consisting on an url and the status code.
+//.
+//# Empty :: Response a
+//.
+//. An empty response with status code 204.
+//.
+//# NotFound :: Response a
+//.
+//. The default NotFound response.
+const Response = Daggy.taggedSum ('Response', {
+  Typed: ['type', 'code', 'value'],
+  Serialized: ['serializer', 'type',  'code', 'value'],
+  Redirect: ['code', 'url'],
+  Empty: [],
+  NotFound: []
+});
+
+function stringify(x) {
+  try {
+    return JSON.stringify (x, null, 2) || String (x);
```

Comment:
`JSON.stringify` will throw, not return falsy, when it fails. So it seems to me as though `|| String(x)` is pretty much dead code (except for when you're stringifying falsy values)

Reply:
This is here just for functions. `JSON.stringify` returns undefined when a function is passed.